code/__DEFINES/loadout.dm

```diff
@@ -22,22 +216,342 @@ GLOBAL_LIST_INIT(loadout_linked_vendor, list(
+	))
+
+///Assoc list linking the job title with their specific clothes vendor
+GLOBAL_LIST_INIT(job_specific_clothes_vendor, list(
+	SQUAD_MARINE = GLOB.marine_clothes_listed_products,
+	SQUAD_ENGINEER = GLOB.engineer_clothes_listed_products,
+	SQUAD_CORPSMAN = GLOB.medic_clothes_listed_products,
+	SQUAD_SMARTGUNNER = GLOB.smartgunner_clothes_listed_products,
+	SQUAD_LEADER = GLOB.leader_clothes_listed_products,
+))
+
+GLOBAL_LIST_INIT(loadout_role_essential_set, list(
+	SQUAD_ENGINEER = list (
+		/obj/item/explosive/plastique = 1,
+		/obj/item/explosive/grenade/chem_grenade/razorburn_smol = 1,
+		/obj/item/stack/sandbags_empty/full = 1,
+		/obj/item/stack/sheet/metal/large_stack = 1,
+		/obj/item/stack/sheet/plasteel/medium_stack = 1,
+		/obj/item/cell/high = 1,
+		/obj/item/tool/shovel/etool = 1,
+		/obj/item/lightreplacer = 1,
+		/obj/item/circuitboard/general = 1,
+	),
+	SQUAD_CORPSMAN = list(
+		/obj/item/bodybag/cryobag = 1,
+		/obj/item/defibrillator = 1,
+		/obj/item/healthanalyzer = 1,
+		/obj/item/roller/medevac = 1,
+		/obj/item/medevac_beacon = 1,
+		/obj/item/roller = 1,
+		/obj/item/tweezers = 1,
+		/obj/item/reagent_containers/hypospray/advanced/oxycodone = 1,
+		/obj/item/storage/firstaid/adv = 1,
+	),
+	SQUAD_SMARTGUNNER = list(
+		/obj/item/clothing/glasses/night/m56_goggles = 1,
+		/obj/item/weapon/gun/rifle/standard_smartmachinegun = 1,
+		/obj/item/ammo_magazine/standard_smartmachinegun = 3,
+	),
+	SQUAD_LEADER = list(
+		/obj/item/explosive/plastique = 1,
+		/obj/item/squad_beacon = 1,
+		/obj/item/squad_beacon = 1,
+		/obj/item/squad_beacon/bomb = 1,
+		/obj/item/whistle = 1,
+		/obj/item/radio = 1,
+		/obj/item/motiondetector = 1,
+		/obj/item/binoculars/tactical = 1,
+		/obj/item/pinpointer/pool = 1,
+	)
+))
+
+///Items that once they are in loadouts, should not have stock checked when sold by the loadout vendor
 GLOBAL_LIST_INIT(bypass_vendor_item, typecacheof(list(
 	/obj/item/ammo_magazine/handful,
 	)))
 
-#define MARINE_LOADOUT "marine"
-#define ENGINEER_LOADOUT "engie"
-#define MEDIC_LOADOUT "medic"
-#define SMARTGUNNER_LOADOUT "smartgunner"
-#define LEADER_LOADOUT "leader"
+///Items that should be saved in loadouts no matter what
+GLOBAL_LIST_INIT(bypass_loadout_check_item, typecacheof(list(
+	/obj/item/clothing/under/marine,
+)))
+
+///List of all supported job titles by the loadout vendors
+GLOBAL_LIST_INIT(loadout_job_supported, list(
+	SQUAD_MARINE,
+	SQUAD_ENGINEER,
+	SQUAD_CORPSMAN,
+	SQUAD_SMARTGUNNER,
+	SQUAD_LEADER,
+))
+	
 
 //Defines use for the visualisation of loadouts
 #define NO_OFFSET "0%"
 #define NO_SCALING 1
 #define MODULAR_ARMOR_OFFSET_Y "-10%"
 #define MODULAR_ARMOR_SCALING 1.2
 
-///The maximum number of loadouts one player can have
-#define MAXIMUM_LOADOUT 50
+///The maximum number of loadouts one player can have per job
+#define MAXIMUM_LOADOUT 15
+
+/// The currently accepted loadout version, all other loadouts will be erased from savefiles
+#define CURRENT_LOADOUT_VERSION 2
```

Comment:
probably can split into major and minor version ? minor versions can update and major versions just straight delete ? dunno if thats possible or needed at all

Reply:
uneeded for now